### ISSUE TITLE
bug: [lua] init.lua:617: phase_func(): failed to receive RPC_HTTP_REQ_CALL: closed

### ISSUE THREAD

**Issue by fmcodeing** [apache/apisix-java-plugin-runner#67]
官方demo StopRequestDemoFilter 请求503,没有正常响应，版本2.9 插件版本0.1

**Comment by tzssangglass** [comment]
see: https://github.com/apache/apisix-java-plugin-runner/blob/main/docs/en/latest/development.md#prerequisites

adapted to APISIX 2.7

**Comment by Hugo-X** [comment]
> 官方demo StopRequestDemoFilter 请求503,没有正常响应，版本2.9 插件版本0.1

Both  APISIX 2.7 and 2.9 can work fine.

A simple modification of the StopRequestDemoFilter.java is needed,  to avoid the **NumberFormatException** or **ClassCastException**

You can try this patch out
https://github.com/Hugo-X/apisix-java-plugin-runner/commit/6eca8cd6710b689dae09b89c5eb4462724be1f21

```
"ext-plugin-pre-req": {
            "conf": [{
                "name": "StopRequestDemoFilter",
                "value": "{\"stop_response_code\": 200, \"stop_response_header_name\": \"header_java_runner\", \"stop_response_header_value\": \"via-java-runner\",  \"stop_response_body\": \"hellox\"}"
            }]
        }
```

```
"ext-plugin-pre-req": {
            "conf": [{
                "name": "StopRequestDemoFilter",
                "value": "{\"stop_response_code\": "200", \"stop_response_header_name\": \"header_java_runner\", \"stop_response_header_value\": \"via-java-runner\",  \"stop_response_body\": \"hellox\"}"
            }]
        }
```


![image](https://user-images.githubusercontent.com/11963765/133880035-ffb384d6-3f76-4ddf-a60c-4785d948e54f.png)

**Comment by fmcodeing** [comment]
thx




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "apache/apisix-java-plugin-runner"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月18日(星期六) 下午3:37
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [apache/apisix-java-plugin-runner] bug: [lua] init.lua:617: phase_func(): failed to receive RPC_HTTP_REQ_CALL: closed (#67)





  
官方demo StopRequestDemoFilter 请求503,没有正常响应，版本2.9 插件版本0.1
  
Both  APISIX 2.7 and 2.9 can work fine.
 
A simple modification of the StopRequestDemoFilter.java is needed,  to avoid the NumberFormatException or ClassCastException
 
You can try this patch out
 ***@***.***
 "ext-plugin-pre-req": {             "conf": [{                 "name": "StopRequestDemoFilter",                 "value": "{\"stop_response_code\": 200, \"stop_response_header_name\": \"header_java_runner\", \"stop_response_header_value\": \"via-java-runner\",  \"stop_response_body\": \"hellox\"}"             }]         }  "ext-plugin-pre-req": {             "conf": [{                 "name": "StopRequestDemoFilter",                 "value": "{\"stop_response_code\": "200", \"stop_response_header_name\": \"header_java_runner\", \"stop_response_header_value\": \"via-java-runner\",  \"stop_response_body\": \"hellox\"}"             }]         }  

 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by tzssangglass** [comment]
No response, considered fixed.
